### PR TITLE
chore: Add useVariation and useVariationDetail hooks for react universal

### PR DESCRIPTION
**Pull request by yusinto**

### CODE REVIEW

File: packages/shared/sdk-client/src/api/LDClient.ts [hunk 1]
```diff
@@ -321,7 +321,7 @@ export interface LDClient {
    * Determines the variation of a feature flag for a context, along with information about how it was
    * calculated.
    *
-   * Note that this will only work if you have set `evaluationExplanations` to true in {@link LDOptions}.
```
Comment (yusinto) [inline packages/shared/sdk-client/src/api/LDClient.ts:324]: This is an existing minor bug in sdk-client.

File: packages/sdk/react-universal/src/client/hooks/variation/useTypedVariation.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import { useLDClient } from '../useLDClient';
+
+export const useTypedVariation = <T extends boolean | number | string | unknown>(
+  key: string,
+  defaultValue: T,
+): T => {
+  const ldClient = useLDClient();
+
+  switch (typeof defaultValue) {
+    // case 'boolean':
```
Comment (kinyoklion) [inline packages/sdk/react-universal/src/client/hooks/variation/useTypedVariation.ts:12]: A assume this is so we can just uncomment it when we switched to the js-core SDK?

File: packages/shared/sdk-client/src/api/LDClient.ts (hunk 1 above)
Comment (kinyoklion) [reply packages/shared/sdk-client/src/api/LDClient.ts:324]: This should be an independent PR for the changelog.

File: packages/sdk/react-universal/src/client/hooks/variation/useTypedVariation.ts (hunk 2 above)
Comment (yusinto) [reply packages/sdk/react-universal/src/client/hooks/variation/useTypedVariation.ts:12]: Yes correct.

File: packages/shared/sdk-client/src/api/LDClient.ts (hunk 1 above)
Comment (kinyoklion) [reply packages/shared/sdk-client/src/api/LDClient.ts]: I went ahead and removed this from this PR and moved it to its own branch,

File: packages/sdk/react-universal/src/ldClientRsc.ts [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
+  LDFlagSet,
+  LDFlagValue,
+} from '@launchdarkly/node-server-sdk';
 
 import { isServer } from './isServer';
 import type { JSSdk } from './types';
 
+// GOTCHA: Partially implement the js sdk.
+// Omit variationDetail because its return type is incompatible with js-core.
+type PartialJSSdk = Omit<Partial<JSSdk>, 'variationDetail'>;
```
[the remark this answers is not in the document]
Comment (kinyoklion) [reply packages/sdk/react-universal/src/ldClientRsc.ts:13]: Added a story: LDEvaluationDetail should have optional reasons for client and required reasons for server.

**Review by kinyoklion — APPROVED** [review]

File: packages/shared/sdk-client/src/api/LDClient.ts (hunk 1 above)
Comment (yusinto) [reply packages/shared/sdk-client/src/api/LDClient.ts]: Apologies I was planning to do this sooner, but got carried away with other work. Thank you for helping.

**Pull request merged**